### PR TITLE
Workspaces: support for building multiple versions

### PR DESCRIPTION
**Pull request by kevireilly**
Using workspaces/packages will allow multiple versions of the web server app to be maintained while the current arrangement only allows 1 version / Vite app to be built. This is mostly in preparation for the upcoming v3 revision with a much improved HA inspired UI.
- Static files for v1 / v2 should still be built and deployed like they are currently
- Adding another version package should be simple after this change set
- Added the ability to override the proxy target via `PROXY_TARGET` environment variable
- Updated documentation for the new arrangement

Related to:
- Working thread: https://discord.com/channels/429907082951524364/1177062706340171807
- Prerequisite for: https://github.com/esphome/esphome-webserver/pull/49
- #49 may also require: https://github.com/esphome/esphome/pull/6203

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -1,32 +1,13 @@
 {
   "name": "esphome-webserver",
-  "version": "2.0.5",
-  "main": "index.js",
```
Comment (kevireilly) [inline package.json:4]: I don't believe this is used unless this repo/package is being published to the NPM registry.

File: package.json [hunk 2]
```diff
@@ -1,32 +1,13 @@
...
+  "version": "3.0.0",
```
Comment (kevireilly) [inline package.json:3]: Though process is this is a relatively significant change to the structure of this repository, though this bump could wait for the upcoming v3 work instead. No requirement to bump it before or after.

File: package.json [hunk 3]
```diff
@@ -1,32 +1,13 @@
...
   "license": "MIT",
+  "workspaces": [
+    "packages/*"
+  ],
```
Comment (kevireilly) [inline package.json:7]: For more information, see: https://docs.npmjs.com/cli/v10/using-npm/workspaces

File: package.json [hunk 4]
```diff
@@ -1,32 +1,13 @@
...
-    "vite-plugin-package-version": "^1.0.2",
-    "vite-plugin-singlefile": "^0.5.1"
-  },
-  "devDependencies": {
-    "@rollup/plugin-node-resolve": "^13.0.6",
-    "@rollup/plugin-replace": "^3.0.0",
-    "@types/node": "^15.12.1",
-    "rollup-plugin-strip-banner": "^2.0.0",
-    "typescript": "^4.1.3",
-    "vite": "^2.3.6"
```
Comment (kevireilly) [inline package.json:30]: All migrated to `packages/v2`

File: package.json [hunk 5]
```diff
@@ -1,32 +1,13 @@
...
+    "build": "npm run build:packages && npm run deploy",
+    "build:packages": "npm run build --workspaces --if-present",
+    "deploy": "npm run deploy --workspaces --if-present"
```
Comment (kevireilly) [inline package.json:11]: Essentially executes `npm run [cmd]` in every workspace that has the `[cmd]` script.

File: packages/v1/package.json [hunk 6]
```diff
@@ -0,0 +1,6 @@
+{
+  "name": "@esphome-webserver/v1",
+  "scripts": {
+    "build": "mkdir -p ../../_static/v1 && cp ./src/* ../../_static/v1/"
+  }
+}
```
Comment (kevireilly) [inline packages/v1/package.json:6]: This conceptually replaces the copying that was previously happening via vite build process.

**Review by kevireilly — COMMENTED** [review]
Related to the following pull requests:
- https://github.com/esphome/esphome-webserver/pull/49
- https://github.com/esphome/esphome/pull/6203
- https://github.com/afarago/esphome-webserver/pull/4

Related to the WIP v3 work mostly being organized here: https://discord.com/channels/429907082951524364/1177062706340171807

cc @afarago @RFDarter @jesserockz @kbx81 @ssieb

**Comment by kevireilly** [comment]
Worth nothing that according to the netlify build/deploy process (which presumably uses this change set), it seems to work just like it did before which is expected behavior.

**Review by jesserockz — APPROVED** [review]
LGTM

**Pull request merged**